students/jsward/session03/strformat_lab.py

```diff
@@ -0,0 +1,58 @@
+# String format lab
+
+from decimal import Decimal
+
+four_elements = (2, 123.4567, 10000, 12345.67)
+
+
+def format_string(e4):
+    # Task 1
+    formatted_string = "file_{:03d} :   {:.2f}, {:.2e}, {:.3g}".format(e4[0], Decimal(e4[1]), e4[2], e4[3])
+    return formatted_string
+
+
+def format_string_kw(e4):
+    # Task 2
+    formatted_string = "file_{pad:03d} :   {dec2:.2f}, {sn2:.2e}, {sf3:.3g}".format(pad=e4[0], dec2=Decimal(e4[1]), sn2=e4[2], sf3=e4[3])
+    return formatted_string
+
+
+def variable_input(elements):
+    # Task 3.  Understanding what the assignment was asking for took longer than writing the code
```

Comment:
I just took another look the exercise, and I can't see how to make it more clear -- suggestions accepted!